### PR TITLE
network/cni: Bring up the `lo` interface for rkt

### LINKED ISSUE #28561: rkt: When using network-plugin=cni, loopback interface is not up
Docker automatically brings up `lo`, but in the rkt case kubelet creates the loopback interface. When using kubenet, it's brought "UP" by kubenet itself now, but with CNI it isn't.

I think we need to do one of the following:
* Add special casing in CNI to call the loopback plugin always (matches what kubenet does)
* Special case calling CNI twice, once always with loopback, once with the configured CNI plugin in the rkt code (special cased for all non-kubenet non-no-op plugins)

cc @kubernetes/sig-rktnetes

### PR DESCRIPTION
**Pull request by euank**
This is already done in kubenet. This specifically fixes an issue where a kubelet-managed network for the rkt runtime does not have an "UP" lo interface.

Fixes #28561

If this fix doesn't seem right, it could also be implemented by rkt effectively managing two "cni" network plugins, one for the user requested network, one for lo.

Followup CRs can improve unit testing further and then possibly remove the vendor directory logic (which seems like dead code)

cc @kubernetes/sig-rktnetes @kubernetes/sig-network @dcbw

### CODE REVIEW

**Comment by euank** [comment]
I'd also like to suggest this as a cherry-pick candidate and possibly re-title to clarify that it ends up impacting rkt+cni and making it a release note / bugfix.

**Comment by dcbw** [comment]
LGTM.  Would be useful for https://github.com/openshift/origin/pull/9981 too since I hard-coded the same thing there (though not with the CNI 'lo' plugin).

File: pkg/kubelet/network/cni/cni.go [hunk 1]
```diff
@@ -87,16 +91,43 @@ func getDefaultCNINetwork(pluginDir, vendorCNIDirPrefix string) (*cniNetwork, er
...
+	return fmt.Sprintf(VendorCNIDirTemplate, prefix, pluginType)
+}
+
+func getLoNetwork(vendorDirPrefix string) *cniNetwork {
+	loConfig, err := libcni.ConfFromBytes([]byte(`{
+  "cniVersion": "0.1.0",
+  "name": "cni-loopback",
+  "type": "loopback"
+}`))
+	if err != nil {
```
Comment (yifan-gu) [inline pkg/kubelet/network/cni/cni.go]: nit: would the error be caused by other things like oom? Though it should be pretty rare.

**Comment by yifan-gu** [comment]
LGTM

File: pkg/kubelet/network/cni/cni.go (hunk 1 above)
Comment (thockin) [inline pkg/kubelet/network/cni/cni.go]: OOm should not manifest as an error ever.

**Comment by thockin** [comment]
@dcbw you should have enough authority to apply the lgtm label here.  You are fully authorized to approve changes like this - you are as much an owner of this code as anyone.

**Comment by fabioy** [comment]
@euank Cherrypick approved. Please create a cherrypick PR. Thanks.

**Comment by euank** [comment]
Can we swap the label here to "release-note"?

**Pull request merged**